dev/bots/analyze.dart

```diff
@@ -197,6 +200,53 @@ Future<void> run(List<String> arguments) async {
 
 // TESTS
 
+/// Verify tool test files end in `_test.dart`.
+///
+/// The test runner will only recognize files ending in `_test.dart` as tests to
+/// be run: https://github.com/dart-lang/test/tree/master/pkgs/test#running-tests
+Future<void> verifyToolTestsEndInTestDart(String workingDirectory) async {
+  final String toolsTestPath = path.join(
+    workingDirectory,
+    'packages',
+    'flutter_tools',
+    'test',
+  );
+  final List<String> violations = <String>[];
+
+  // detect files that contains calls to test(), testUsingContext(), and testWithoutContext()
+  final RegExp callsTestFunctionPattern = RegExp(r'(test\(.*\)|testUsingContext\(.*\)|testWithoutContext\(.*\))');
+
+  await for (final File file in _allFiles(toolsTestPath, 'dart', minimumMatches: 42)) {
```

Comment:
Remind me, what does `minimumMatches` do?

Reply:
oops, I need to change this. I put 42 there to get this working, lol

Reply:
I'm not sure the rationale (I asked in #hacker-tests) but what it effectively does is assert that it found more than this before returning. I upped this to 300 as we currently have 357 test files).